hedera-node/hedera-mono-service/src/main/java/com/hedera/services/state/virtual/entities/OnDiskAccount.java

```diff
@@ -0,0 +1,947 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.state.virtual.entities;
+
+import static com.hedera.services.context.properties.StaticPropertiesHolder.STATIC_PROPERTIES;
+import static com.hedera.services.legacy.core.jproto.JKey.equalUpToDecodability;
+import static com.hedera.services.legacy.proto.utils.ByteStringUtils.unwrapUnsafelyIfPossible;
+import static com.hedera.services.legacy.proto.utils.ByteStringUtils.wrapUnsafely;
+import static com.hedera.services.state.merkle.internals.BitPackUtils.codeFromNum;
+import static com.hedera.services.state.submerkle.EntityId.MISSING_ENTITY_ID;
+import static com.hedera.services.state.virtual.KeyPackingUtils.*;
+import static com.hedera.services.state.virtual.utils.EntityIoUtils.readBytes;
+import static com.hedera.services.state.virtual.utils.EntityIoUtils.writeBytes;
+import static com.hedera.services.utils.MiscUtils.asFcKeyUnchecked;
+import static com.hedera.services.utils.SerializationUtils.*;
+
+import com.google.protobuf.ByteString;
+import com.hedera.services.exceptions.NegativeAccountBalanceException;
+import com.hedera.services.legacy.core.jproto.JKey;
+import com.hedera.services.legacy.core.jproto.JKeySerializer;
+import com.hedera.services.state.merkle.MerkleAccountState;
+import com.hedera.services.state.migration.HederaAccount;
+import com.hedera.services.state.submerkle.EntityId;
+import com.hedera.services.state.submerkle.FcTokenAllowanceId;
+import com.hedera.services.state.virtual.ContractKey;
+import com.hedera.services.state.virtual.annotations.StateSetter;
+import com.hedera.services.state.virtual.utils.CheckedConsumer;
+import com.hedera.services.state.virtual.utils.CheckedSupplier;
+import com.hedera.services.utils.EntityNum;
+import com.hedera.services.utils.EntityNumPair;
+import com.hederahashgraph.api.proto.java.Key;
+import com.hederahashgraph.api.proto.java.KeyList;
+import com.swirlds.common.io.streams.SerializableDataInputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.virtualmap.VirtualValue;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.*;
+import org.jetbrains.annotations.Nullable;
+
+public class OnDiskAccount implements VirtualValue, HederaAccount {
+    private static final int CURRENT_VERSION = 1;
+    private static final long CLASS_ID = 0xc88e3a5c7b497468L;
+
+    private static final String EMPTY_MEMO = "";
+    private static final JKey HOLLOW_KEY;
+
+    static {
+        HOLLOW_KEY =
+                asFcKeyUnchecked(Key.newBuilder().setKeyList(KeyList.getDefaultInstance()).build());
+    }
```

Comment:
Do we set some default key on every account instead of `null` or `Optional.empty`?

Reply:
We _should_ always set a default key right now; but yeah, I added this b/c it seemed like it would be useful for lazy account creation. 🙂

Reply:
Got it ! Thanks Michael !